### DISCUSSION TITLE
Multiple WSGI mounts

### DISCUSSION THREAD

**Discussion by JoseMoreville** [fastapi/fastapi#7067] [Questions] [answered]
Hello, I'm new to fastapi and im trying to use WSGIMiddleware to connect to a full flask app (With multiple routes) and so far i can just access to it's main route.

i'm using app.mount to run my app as "/menu" but it just run "/" set on flask
`app.mount("/menu", WSGIMiddleware(flask_app))`

In case you need i can provide more code or even i can explain in spanish if you need more in-depth explanation.

**Comment by Mause** [comment]
Can you provide a code sample that demonstrates the failure?

**Comment by JoseMoreville** [comment]
> Can you provide a code sample that demonstrates the failure?
```
from fastapi import FastAPI
from fastapi.middleware.wsgi import WSGIMiddleware
from flask import Flask

flask_app = Flask(__name__)
app = FastAPI()


@flask_app.route('/')
@mobile_template('{mobile/}menu/menu.html')
def home(template):
    return render_template(template,
                           response=response,
                           response=response)

@flask_app.route('/quiz', methods=["GET", "POST"])
@mobile_template('{mobile/}app_index.html')
def quiz(template):
    return make_response(render_template(template,
                                             dict_options=icons_dict,
                                             case_id=case_id,
                                             total_cases=total_cases,
                                             ), 200)



@app.get("/menu", response_class=HTMLResponse)
async def home(request: Request):
    return templates.TemplateResponse("index.html", {"request": request})


app.mount("/v1", WSGIMiddleware(flask_app))
```

All fastAPI routes on my code work well just flask mount is not working as i want to. Menu should be able to move to quiz but it doesn't work nor the route '/quiz' or '/menu/quiz'

**Comment by JoseMoreville** [comment]
I forgot to add i get this on the rendered page
![image](https://user-images.githubusercontent.com/47700212/100375852-b109ad00-2fcb-11eb-9ea2-bfdc0aa7b955.png)

**Comment by ArcLightSlavik** [comment]
The error is probably in your Flask app, can you check that it works by itself / what the error is (the terminal).
I ran this with template in `templates/index.html` and it works. 

```
from fastapi import FastAPI
from fastapi.middleware.wsgi import WSGIMiddleware

from flask_mobility import Mobility
from flask_mobility.decorators import mobile_template

from flask import Flask, render_template

flask_app = Flask(__name__)
Mobility(flask_app)


@flask_app.route("/")
@mobile_template('index.html')
def index(template):
    return render_template(template)


app = FastAPI()


@app.get("/v2")
def read_main():
    return {"message": "Hello World"}


app.mount("/v1", WSGIMiddleware(flask_app))
```

**Comment by JoseMoreville** [comment, marked as answer]
> The error is probably in your Flask app, can you check that it works by itself / what the error is (the terminal).
> I ran this with template in `templates/index.html` and it works.
> 
> ```
> from fastapi import FastAPI
> from fastapi.middleware.wsgi import WSGIMiddleware
> 
> from flask_mobility import Mobility
> from flask_mobility.decorators import mobile_template
> 
> from flask import Flask, render_template
> 
> flask_app = Flask(__name__)
> Mobility(flask_app)
> 
> 
> @flask_app.route("/")
> @mobile_template('index.html')
> def index(template):
>     return render_template(template)
> 
> 
> app = FastAPI()
> 
> 
> @app.get("/v2")
> def read_main():
>     return {"message": "Hello World"}
> 
> 
> app.mount("/v1", WSGIMiddleware(flask_app))
> ```

I fixed it today, the problem was that i was accessing to get instead of post on my flask routes which is necesary for sessionStorage if i dont have a sessionstorage it won't work.
I just changed the fetch route to '/menu/get'

**Comment by raphaelauv** [comment]
@JoseMoreville could you please close the issue, thank you

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
